viral/representational_drift.py

```
@@ -0,0 +1,828 @@
+    cell_by_cell_correlation = np.empty((n_cells, n_trials, n_trials))
+
+    for neuron_idx in range(n_cells):
+        for i in range(trial_array.shape[0]):
+            for j in range(trial_array.shape[0]):
+                cell_by_cell_correlation[neuron_idx, i, j] = stats.pearsonr(
+                    trial_array[i, neuron_idx, :], trial_array[j, neuron_idx, :]
+                ).correlation
+
+    if plot:
+        plt.plot(
+            basic_normalise(speed_correlation[trial_group_size:]),
```

Ruff (0.14.4)

Undefined name `trial_group_size`

(F821)